src/__tests__/spectral.jest.test.ts

```diff
@@ -0,0 +1,123 @@
+
+jest.mock(
+  'node-fetch',
+  (() => {
+    // I know it might look weird, and it in fact is, so let me explain.
+    // There was some weird issue I had to mitigate against.
+    // Basically, if you deferred loading of fetch-mock module, sandbox method was no longer there for some reason.
+    const fetchMock = require('fetch-mock');
+    return () => fetchMock.sandbox();
+  })(),
+);
+
+describe('Spectral', () => {
+  describe('loadRuleset', () => {
+    test('should support loading built-in rulesets', async () => {
+      const s = new Spectral();
+      await s.loadRuleset('spectral:oas2');
+
+      expect(s.rules).toEqual(
+        [...Object.entries(oasRuleset.rules), ...Object.entries(oas2Ruleset.rules)].reduce<Dictionary<unknown>>(
+          (oasRules, [name, rule]) => {
+            oasRules[name] = {
+              name,
+              ...rule,
+              severity: expect.any(Number),
+            };
+
+            return oasRules;
+          },
+          {},
+        ),
+      );
+    });
+
+    test('should support loading multiple built-in rulesets', async () => {
+      const s = new Spectral();
+      await s.loadRuleset('spectral:oas2', 'spectral:oas3');
+
+      expect(s.rules).toEqual(
+        [
+          ...Object.entries(oasRuleset.rules),
+          ...Object.entries(oas2Ruleset.rules),
+          ...Object.entries(oas3Ruleset.rules),
+        ].reduce<Dictionary<unknown>>((oasRules, [name, rule]) => {
+          oasRules[name] = {
+            name,
+            ...rule,
+            severity: expect.any(Number),
+          };
+
+          return oasRules;
+        }, {}),
+      );
+    });
+
+    test('should support loading rulesets from filesystem', async () => {
+      const s = new Spectral();
+      await s.loadRuleset(path.join(__dirname, '__fixtures__/custom-oas-ruleset.json'));
+
+      expect(s.rules).toEqual({
+        ...[...Object.entries(oasRuleset.rules)].reduce<Dictionary<unknown>>((oasRules, [name, rule]) => {
+          oasRules[name] = {
+            name,
+            ...rule,
+            severity: expect.any(Number),
+          };
+
+          return oasRules;
+        }, {}),
+        'info-matches-stoplight': {
+          ...customOASRuleset.rules['info-matches-stoplight'],
+          name: 'info-matches-stoplight',
+          severity: DiagnosticSeverity.Warning,
+        },
+      });
+    });
+
+    test('should support loading rulesets over http', async () => {
+      const ruleset = {
+        rules: {
+          'info-matches-stoplight': {
+            message: 'Info must contain Stoplight',
+            given: '$.info',
+            type: 'style',
+            then: {
+              field: 'title',
+              function: 'pattern',
+              functionOptions: {
+                match: 'Stoplight',
+              },
+            },
+          },
+        },
+      };
+
+      (fetch as FetchMockSandbox).mock('https://localhost:4000/custom-ruleset', {
```

Comment:
We already use [nock](https://github.com/nock/nock), can we use this instead of fetch-sandbox? 

```cat package.json | grep nock
    "nock": "~10.0.6",```

Or can we switch that nock usage to fetch sandbox?

Either way these seem to do the same sort of thing and we should not have both.

Reply:
Yeah, I was aware of `nock` in the project.
The main obstacle with usage of `nock` is that it's built on top of Node.js and hence we cannot mock requests in a browser using it.
That said, I can convert that Jest test to use `nock`, but will need to leave that mocking solution for Karma-based (browser) tests.
Does it sound good?

Side note,
I planned to share those tests initially, hence I used fetch-mock in both, but then realized I cannot do that in the way I want to and decide to keep them separate (jest and karma).